### PR TITLE
Python 3

### PR DESCRIPTION
**Pull request by davidtaylorhq**
Depends on the reversion update PR, we should merge that first.

### CODE REVIEW

**Comment by davidtaylorhq** [comment]
Now rebased onto master

**Comment by davidtaylorhq** [comment]
@tomtom5152 do you think you're likely to be able to review this before term starts? If not, I suggest we just merge it and see what happens.

**Review by tomtom5152 — APPROVED** [review]
Having given the diff a good scouring and looked for the changes I thought most likely to break, I am happy to approve these for production, certainly with the potential of a revert if it all goes wrong.

That said, I have been unable to send an event approval email from the deploy preview, I suspect this is more a config issue than an actual code issue though, probably an error with exchange (it usually is).

**Pull request merged**
